api/v1beta1/cinderapi_types.go

```diff
@@ -58,6 +58,10 @@ type CinderAPISpec struct {
 	// +kubebuilder:validation:Optional
 	// ExtraMounts containing conf files and credentials
 	ExtraMounts []CinderExtraVolMounts `json:"extraMounts,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// ServiceAccount - service account name used internally to provide Cinder services the default SA name
+	ServiceAccount string `json:"serviceAccount"`
```

Comment:
Sorry that I'm late to the party (code merged already), but I wonder if these common fields should be included in the CinderServiceTemplate struct. We've been trying to dedup a lot of redundant data structures and code. @fmount I think you are more familiar with this effort, and am curious to hear your thoughts.

Reply:
@ASBishop in theory you're right in terms of data structure it can easily be absorbed in  `CinderServiceTemplate`. However, given it's not a field populated by the `Spec` (but the result of `RbacResourceName()` function), we should properly populate this field when the Template is generated.
This was the reason `DatabaseHostname` and a few other fields are still part of the Spec instead of being passed from the upper layer, but I agree we can follow up providing a unified strategy that allows more dedup.